app/models/purchase.rb

```diff
@@ -38,6 +38,7 @@ class Purchase < ApplicationRecord
       .includes(:line_items, :storage_location)
       .order(created_at: :desc)
   }
+  scope :for_year, ->(year) { where("extract(year from issued_at) = ?", year) }
```

Comment:
Three times a charm!

You should be able to move this scope into `IssuedAt` :+1: